src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -227,5 +228,26 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, HttpsConn
 
             return listenOptions;
         }
+
+        /// <summary>
+        /// Configure Kestrel to use HTTPS.
+        /// </summary>
+        /// <param name="listenOptions">The <see cref="ListenOptions"/> to configure.</param>
+        /// <param name="serverOptionsCallback">Callback to configure HTTPS options.</param>
+        /// <param name="state">State for the <see cref="ServerOptionsSelectionCallback" />.</param>
+        /// <returns>The <see cref="ListenOptions"/>.</returns>
+        internal static ListenOptions UseHttps(this ListenOptions listenOptions, ServerOptionsSelectionCallback serverOptionsCallback, object state = null)
```

Comment:
Consider making this public. We want the config to use a consistent API with whatever the eventual public API for this callback is.